src/Microsoft.ML.Transforms/Text/WordEmbeddingsTransform.cs

```diff
@@ -723,60 +725,53 @@ private Model GetVocabularyDictionary()
                     }
                 }
 
-                Model model = null;
-                using (StreamReader sr = File.OpenText(_modelFileNameWithPath))
+                using (var ch = Host.Start(LoaderSignature))
+                using (var pch = Host.StartProgressChannel("Building Vocabulary from Model File for Word Embeddings Transform"))
                 {
-                    string line;
-                    int lineNumber = 1;
-                    char[] delimiters = { ' ', '\t' };
-                    using (var ch = Host.Start(LoaderSignature))
-                    using (var pch = Host.StartProgressChannel("Building Vocabulary from Model File for Word Embeddings Transform"))
-                    {
-                        var header = new ProgressHeader(new[] { "lines" });
-                        pch.SetHeader(header, e => e.SetProgress(0, lineNumber));
-                        string firstLine = sr.ReadLine();
-                        while ((line = sr.ReadLine()) != null)
+                    var parsedData = new ConcurrentBag<(string key, float[] values, long lineNumber)>();
+                    int skippedLinesCount = Math.Max(1, _linesToSkip);
+
+                    Parallel.ForEach(File.ReadLines(_modelFileNameWithPath).Skip(skippedLinesCount),
```

Comment:
>Parallel.ForEach [](start = 20, length = 16)

Please respect Host.ConcurrencyFactor and set MaxDegreeOfParallelism to it's value.

Reply:
@Ivanidzo4ka great catch! thanks! #done